rp/src/exchange.rs

```diff
@@ -0,0 +1,234 @@
+            .up()
+            .execute()
+            .await?;
+
+        Ok(link.header.index)
+    }
+
+    pub async fn link_cleanup(rtnetlink: &Handle, index: u32) -> Result<()> {
+        rtnetlink.link().del(index).execute().await?;
+
+        Ok(())
+    }
+
+    pub async fn link_cleanup_standalone(index: u32) -> Result<()> {
+        let (connection, rtnetlink, _) = rtnetlink::new_connection()?;
+        tokio::spawn(connection);
+
+        // We don't care if this fails, as the device may already have been auto-cleaned up.
+        let _ = rtnetlink.link().del(index).execute().await;
+
+        Ok(())
+    }
+
+    pub async fn wg_set(
+        genetlink: &mut GenetlinkHandle,
+        index: u32,
+        mut attr: Vec<WgDeviceAttrs>,
+    ) -> Result<()> {
+        use futures_util::StreamExt as _;
+        use netlink_packet_core::{NetlinkMessage, NetlinkPayload};
+        use netlink_packet_generic::GenlMessage;
+        use netlink_packet_wireguard::{Wireguard, WireguardCmd};
+
+        attr.insert(0, WgDeviceAttrs::IfIndex(index));
+
+        let wgc = Wireguard {
+            cmd: WireguardCmd::SetDevice,
+            nlas: attr,
+        };
+
+        let genl = GenlMessage::from_payload(wgc);
+        let mut nlmsg = NetlinkMessage::from(genl);
+        nlmsg.header.flags = NLM_F_REQUEST | NLM_F_ACK;
+
+        let (res, _) = genetlink.request(nlmsg).await?.into_future().await;
+        if let Some(res) = res {
+            let res = res?;
+            if let NetlinkPayload::Error(err) = res.payload {
+                return Err(err.to_io().into());
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(any(target_os = "linux", target_os = "freebsd"))]
+pub async fn exchange(options: ExchangeOptions) -> Result<()> {
+    use std::fs::{self, read_to_string};
+
+    use anyhow::anyhow;
+    use base64::Engine;
+    use netlink_packet_wireguard::{constants::WG_KEY_LEN, nlas::WgDeviceAttrs};
+    use rosenpass::{
+        app_server::{AppServer, WireguardOut},
+        config::Verbosity,
+        protocol::{SPk, SSk, SymKey},
+    };
+    use rosenpass_util::file::{LoadValue as _, LoadValueB64};
+    use zeroize::Zeroize;
+
+    let (connection, rtnetlink, _) = rtnetlink::new_connection()?;
+    tokio::spawn(connection);
+
+    let link_name = options.dev.unwrap_or("rosenpass0".to_string());
+    let link_index = netlink::link_create_and_up(&rtnetlink, link_name.clone()).await?;
+
+    ctrlc_async::set_async_handler(async move {
+        netlink::link_cleanup_standalone(link_index)
+            .await
+            .expect("Failed to clean up");
+    })?;
+
+    // Deploy the classic wireguard private key
+    let (connection, mut genetlink, _) = genetlink::new_connection()?;
+    tokio::spawn(connection);
+
+    let wgsk_path = options.private_keys_dir.join("wgsk");
+    let mut wgsk: [u8; WG_KEY_LEN] = base64::engine::general_purpose::STANDARD
+        .decode(&read_to_string(wgsk_path)?)?
```

Comment:
Unfortunately, I am not sure if this is acceptable. While we erase the decoded base64 secret key, we apparently do not do this with the encoded version, in fact it is just a temporary that is immediately discarded.

And what if the base64 crate does some copying internally ... Is there a cryptographically secure base64 crate? Sorry by the way for my pickiness here, I am just a bit paranoid. 👀 

@koraa I would like your feedback on this.

Edit: I see that this crate can be compiled with `no_std`. I think this is a good idea, because we then ensure that the heap is at least not being used.